world-explore/src/app/home/home.page.html

```diff
@@ -10,20 +10,20 @@
         {{ showFavorites ? 'All Countries' : 'View Favorites' }}
       </ion-button>
     </ion-buttons>
-    <ion-searchbar [(ngModel)]="searchTerm" (ionInput)="filterCountries()"></ion-searchbar>
+    <ion-searchbar (ionInput)="filterCountries($event)"></ion-searchbar>
   </ion-toolbar>
 </ion-header>
 
 <ion-content #content>
-  <ion-list *ngFor="let letter of getKeys(groupedCountries)">
+  <ion-list *ngFor="let letter of getKeys(filteredCountries | groupByLetter)">
     <ion-item-divider id="letter-{{ letter }}">
       <ion-label>{{ letter }}</ion-label>
     </ion-item-divider>
-    <ion-item-sliding *ngFor="let country of groupedCountries[letter] | sortCountries"
-      [class.favorite]="favoriteCountries.has(country.country)" class="country-item" (ionSwipe)="toggleFavorite(country, $event)">
+    <ion-item-sliding *ngFor="let country of (filteredCountries | groupByLetter)[letter] | sortCountries"
+      [ngClass]="{ 'favorite': favoriteCountries.has(country.country) }" class="country-item" (ionSwipe)="toggleFavorite(country, $event)">
       <ion-item>
         <ion-thumbnail slot="start">
-          <img [src]="countryFlags[country.country]" *ngIf="countryFlags[country.country]" class="country-flag" />
+          <ion-img [src]="getSafeFlagUrl(country.country)" *ngIf="getSafeFlagUrl(country.country)" class="country-flag"></ion-img>
```

Comment:
wrap ion-img with ng-container  for codition *ngIf="getSafeFlagUrl(country.country)"